### DISCUSSION TITLE
abilitazione OTB

### DISCUSSION THREAD

**Discussion by Forrest-It** [iisgiua/giuaschool#389] [Richieste di aiuto] [answered]
Buongiorno,

volevo chiedere se l'utilizzo OTP è una funzionalità già implementata in giua@school e se qualcuno l'ha già utilizzata.

Grazie

Tommaso

**Comment by trinko** [comment, marked as answer]
Si, l'OTP è funzionante. Per utilizzarlo:
- da amministratore, imposta il parametro **otp_tipo** indicando la lista dei ruoli che sono abilitati (vedi l'impostazione per i dettagli)
- a questo punto gli utenti con ruolo abilitato possono, dal loro profilo, attivare l'utilizzo dell'OTP, che verrà chiesto per ogni login.
- nel caso di problemi, da amministratore, andare su **MODIFICA > DOCENTI** per resettare l'OTP, disattivandolo

Si consiglia **non** di attivare l'OTP per l'amministratore, in modo che non possa essere bloccato in caso di problemi.

Per utilizzare l'OTP, bisogna installare sul proprio smartphone il Google Authenticator.
I problemi tipici sono:
- disallineamento tra Google e il server del registro: capita ogni tanto, ma si risolve sempre tramite l'app Google Authenticator con l'apposita funzione di risincronizzazione;
- rottura/dismissione dello smartphone con installato il Google Authenticator: non si possono trasferire le impostazioni su un altro smartphone, ma è necessario resettare l'OTP da amministratore e poi procedere a riattivarlo di nuovo sul nuovo dispositivo.

**Reply by Forrest-It** [reply]
Salve,

ho seguito le sue istruzioni ( ho abilitato l'OTP sul profilo Genitori da amministraore ) : non riesco a capire nel profilo genitore dove si attivi l'OTP ( da amministratore vedo che per  profilo genitore si può attivare lo SPID  - dal profilo GENITORE non trovo la sezione dove si abililat l'OTP ).

Grazie

Tommaso

**Comment by trinko** [comment]
In effetti, lato genitore, sono rimaste le vecchie pagine del profilo che impediscono l'attivazione dell'OTP. 
In origine, infatti, era stato impostato solo per i docenti, e per loro dovrebbe funzionare correttamente. 

La nuova gestione, che prevede l'utilizzo dell'OTP per gli utenti scelti lato amministratore, non è stata completata per mancanza di tempo.

**Comment by sempliciniluigi89** [comment]
Quindi nella prossima versione sarà possibile scegliere dal lato
amministratore l'OTP ?

Il giorno ven 7 ott 2022 alle ore 18:25 Antonello Dessì <
***@***.***> ha scritto:

**Comment by sempliciniluigi89** [comment]
Lo chiedo perché dal lato amministratore in parametri ho provato ad abilitare l'otp a tutti gli utenti e non succede niente ,stessa cosa dicasi se lo faccio per singola categoria di utenti ,loro riescono ad entrare normalmente con username e password senza aver nessun effetto sull'otp abilitato in segreteria

**Reply by sempliciniluigi89** [reply]
L'unico utente a cui va a buon fine è l'utenza docente

**Comment by trinko** [comment]
> Quindi nella prossima versione sarà possibile scegliere dal lato amministratore l'OTP ? 

No, in una delle prossime versioni. 
Attualmente l'OTP non è una priorità... Non si può fare tutto...
